File: python/meteroid/models/checkout_session.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .checkout_session_id import CheckoutSessionId
from .checkout_session_status import CheckoutSessionStatus
from .checkout_type import CheckoutType
from .customer_id import CustomerId
from .payment_methods_config import PaymentMethodsConfig
from .plan_version_id import PlanVersionId
from .subscription_id import SubscriptionId


@dataclasses.dataclass
class CheckoutSession(BaseModel):
    checkout_type: CheckoutType

    created_at: datetime

    customer_id: CustomerId

    id: CheckoutSessionId

    plan_version_id: PlanVersionId

    status: CheckoutSessionStatus

    billing_day_anchor: t.Optional[int] = None

    billing_start_date: t.Optional[str] = None

    cancel_url: t.Optional[str] = None

    checkout_url: t.Optional[str] = None

    completed_at: t.Optional[datetime] = None

    coupon_code: t.Optional[str] = None

    expires_at: t.Optional[datetime] = None
    """When the session expires. None means the session never expires."""

    net_terms: t.Optional[int] = None

    payment_methods_config: t.Optional[PaymentMethodsConfig] = None

    subscription_id: t.Optional[SubscriptionId] = None

    success_url: t.Optional[str] = None

    trial_duration_days: t.Optional[int] = None

File: go/checkout_session.go
// this file is @generated
package meteroid

import "time"

type CheckoutSession struct {
	BillingDayAnchor *int32 `json:"billing_day_anchor,omitempty"`

	BillingStartDate *string `json:"billing_start_date,omitempty"`

	CheckoutType CheckoutType `json:"checkout_type"`

	CheckoutUrl *string `json:"checkout_url,omitempty"`

	// RFC 3339 timestamp.
	CompletedAt *time.Time `json:"completed_at,omitempty"`

	CouponCode *string `json:"coupon_code,omitempty"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	CustomerId CustomerId `json:"customer_id"`

	// When the session expires. None means the session never expires.
	// RFC 3339 timestamp.
	ExpiresAt *time.Time `json:"expires_at,omitempty"`

	Id CheckoutSessionId `json:"id"`

	NetTerms *int32 `json:"net_terms,omitempty"`

	PaymentMethodsConfig *PaymentMethodsConfig `json:"payment_methods_config,omitempty"`

	PlanVersionId PlanVersionId `json:"plan_version_id"`

	Status CheckoutSessionStatus `json:"status"`

	SubscriptionId *SubscriptionId `json:"subscription_id,omitempty"`

	TrialDurationDays *int32 `json:"trial_duration_days,omitempty"`
}
